universal-application-tool-0.0.1/app/models/Question.java

```diff
@@ -33,6 +33,19 @@
 import services.question.types.QuestionDefinitionBuilder;
 import services.question.types.QuestionType;
 
+/**
+ * An EBean mapped class that stores the configuration for a single question.
+ *
+ * <p>Questions can be shared by multiple {@code Program}s. The set of questions for a given program
+ * define the data model for a given applicant i.e. all of the data that can be collected for an
+ * {@code Applicant} across all programs.
+ *
+ * <p>After they are published questions are immutable. To update a question the CiviForm admin must
+ * create a new {@code Version} draft, make updates, and publish which will create a new version of
+ * all updated questions.
+ *
+ * <p>Questions that aren't updated between versions are associated with multiple versions.
```

Comment:
praise: this comment is very helpful! Would you also add this to program? A program that is not updated will be carried over to a new version.

Reply:
Done